internal/framework5provider/schema_ephemeral_resource_test.go

```diff
@@ -0,0 +1,750 @@
+					statecheck.ExpectKnownValue(
+						"echo.schema_test",
+						echoDataPath.AtMapKey("dynamic_attribute"),
+						knownvalue.TupleExact([]knownvalue.Check{
+							knownvalue.Int64Exact(12),
+							knownvalue.Int64Exact(34),
+						}),
+					),
+					statecheck.ExpectKnownValue("echo.schema_test", echoDataPath.AtMapKey("float32_attribute"), knownvalue.Null()),
+					statecheck.ExpectKnownValue("echo.schema_test", echoDataPath.AtMapKey("float64_attribute"), knownvalue.Null()),
+					statecheck.ExpectKnownValue("echo.schema_test", echoDataPath.AtMapKey("int32_attribute"), knownvalue.Null()),
+					statecheck.ExpectKnownValue("echo.schema_test", echoDataPath.AtMapKey("int64_attribute"), knownvalue.Null()),
+					statecheck.ExpectKnownValue("echo.schema_test", echoDataPath.AtMapKey("list_attribute"), knownvalue.Null()),
+					statecheck.ExpectKnownValue("echo.schema_test", echoDataPath.AtMapKey("list_nested_block"), knownvalue.ListSizeExact(0)),
+					statecheck.ExpectKnownValue("echo.schema_test", echoDataPath.AtMapKey("map_attribute"), knownvalue.Null()),
+					statecheck.ExpectKnownValue("echo.schema_test", echoDataPath.AtMapKey("number_attribute"), knownvalue.Null()),
+					statecheck.ExpectKnownValue("echo.schema_test", echoDataPath.AtMapKey("object_attribute"), knownvalue.Null()),
+					statecheck.ExpectKnownValue("echo.schema_test", echoDataPath.AtMapKey("set_attribute"), knownvalue.Null()),
+					statecheck.ExpectKnownValue("echo.schema_test", echoDataPath.AtMapKey("set_nested_block"), knownvalue.ListSizeExact(0)),
+					statecheck.ExpectKnownValue("echo.schema_test", echoDataPath.AtMapKey("single_nested_block"), knownvalue.Null()),
+					statecheck.ExpectKnownValue("echo.schema_test", echoDataPath.AtMapKey("string_attribute"), knownvalue.Null()),
+				},
+			},
+		},
+	})
+}
+
+func TestSchemaEphemeralResource_Float32Attribute(t *testing.T) {
+	t.Parallel()
+
+	resource.UnitTest(t, resource.TestCase{
+		// Ephemeral resources are only available in 1.10 and later
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.SkipBelow(tfversion.Version1_10_0),
+		},
+		ProtoV5ProviderFactories: map[string]func() (tfprotov5.ProviderServer, error){
+			"framework": providerserver.NewProtocol5WithError(New()),
+		},
+		ProtoV6ProviderFactories: map[string]func() (tfprotov6.ProviderServer, error){
+			"echo": echoprovider.NewProviderServer(),
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: addEchoToEphemeralSchemaConfig(`ephemeral "framework_schema" "test" {
+					float32_attribute = 1234.5
+				}`),
+				ConfigStateChecks: []statecheck.StateCheck{
+					statecheck.ExpectKnownValue("echo.schema_test", echoDataPath.AtMapKey("bool_attribute"), knownvalue.Null()),
+					statecheck.ExpectKnownValue("echo.schema_test", echoDataPath.AtMapKey("dynamic_attribute"), knownvalue.Null()),
+					statecheck.ExpectKnownValue("echo.schema_test", echoDataPath.AtMapKey("float32_attribute"), knownvalue.Float32Exact(1234.5)),
```

Comment:
nit: I typically move the actual test assertion to the top so that you can skim over the null assertions. Although that will mean that the assertion order is different from test to test, so I don't have a strong opinion if you want to keep it in alphabetical order.

Reply:
I can vibe with that 👍🏻 -> adjusted in 046f43cb0bb6720dfbdbb0aafa1d7979a1c05bea (as well as fixed the RC1 tests that were expected to break now that our CI is running it 👍🏻 ) ⚡